stores/filestore.go

```diff
@@ -546,6 +545,9 @@ type FileSubStore struct {
 	activity    bool         // was there any write between two flush calls
 	writer      io.Writer    // this is either `bw` or `file` depending if buffer writer is used or not
 	shrinkTimer *time.Timer  // timer associated with callback shrinking buffer when possible
+	syncTimer   *time.Timer  // timer associated with performing auto flush and disk sync
+	needSync    bool         // this is required to reduce sync'ing in case DoSync==false, but AutoSync>0
+	synced      int64        // number of times the file is actually sync'ed
```

Comment:
Why do we need this? Just curious.

Reply:
You mean `synced` count? This is mainly to check in tests that we don't sync when not needed.

Reply:
ok